lib/chef/resource/kernel_module.rb

```diff
@@ -68,13 +68,21 @@ class KernelModule < Chef::Resource
         end
         ```
 
-        Disable a kernel module.
+        Disable a kernel module so that it is uninstallable.
```

Comment:
```suggestion
        Disable a kernel module so that it can be uninstalled.
```

Reply:
That's in accurate, it becomes uninstallable after you run the `:disable` action.

The resource,

```ruby
kernel_module 'foo' do
  action :disable
end
```
creates a "fake installation"

```bash
# /etc/modprobe.d/disable_foo.conf
install foo /bin/false
```

https://github.com/chef/chef/blob/main/lib/chef/resource/kernel_module.rb#L174

Which prevents the insertion of the module, e.g.:

```bash
[root@ip-172-0-0-1 ~]# modprobe foo
modprobe: ERROR: Error running install command for foo
modprobe: ERROR: could not insert 'foo': Operation not permitted
[root@ip-172-0-0-1 ~]# rm /etc/modprobe.d/disable_foo.conf 
rm: remove regular file ‘/etc/modprobe.d/disable_foo.conf’? y
[root@ip-172-0-0-1 ~]# modprobe foo
modprobe: FATAL: Module foo not found.
```

Even though `foo` doesn't exist, AND because of the `install foo /bin/false`, modprobe doesn't both to check if the foo exists because it knows it's uninstallable anyway.

Reply:
do you mean that it is "not installable" after disabling it?

Reply:
bahahaha, yes!  Yours is way better!